spec/src/main/asciidoc/required-apis.asciidoc

```diff
@@ -41,11 +37,17 @@ MicroProfile applications that make use of APIs or versions not mentioned below
  - <<mp-opentracing, MicroProfile OpenTracing 2.0>>
  - <<mp-rest-client, MicroProfile Rest Client 2.0>>
 
+A MicroProfile 4.x compatible implementation must pass all the TCKs provided by the MicroProfile individual
+specifications. The Jakarta specifications TCKs are not required.
+
+These APIs and versions guarantee application developers a minimum set of APIs and are not intended to be restrictive.
+MicroProfile applications that make use of APIs or versions not mentioned here are not considered portable.
```

Comment:
What does portable mean in this context?

Reply:
My interpretation of this (and this text was already in the spec, I just moved it around), is that if an application uses any API (not an implementation, for instance Bean Validation), that is not on the list then the application may not be able to run in all MP available runtimes.

Reply:
I would delete the whole sentence "MicroProfile applications that make use of APIs or versions not mentioned here are not considered portable." as the previous sentence has hinted. Besides, they might be portable if they move among the runtimes that support the same.

Reply:
Sure; and I don't want to hijack the pull request.  But for "compatible implementation" purposes I do think _somewhere_ it is important to say something rather in depth about portability. For example, there is no world that I'm aware of in which a Helidon application will run on or in OpenLiberty, and no world in which a Payara application will run in Quarkus, and so on etc. If that's OK, as the intent behind this compatibility effort seems to be, then users are probably owed a paragraph or two explaining the kind of portability MicroProfile is talking about.

Reply:
@ljnelson do you think we should just remove this part?

Reply:
I find this section somewhat confusing, tbh. Regarding the platform spec, I would say a better approach is "The behavior and APIs encompassed by this specification are considered consistent across compatible implementations." OK, not the best wording, but using the word "portable" doesn't really fit. Interestingly, we use the phrase "Platform Spec" when MicroProfile isn't really a standalone platform because of its relatively narrow focus. If we really want to use the word "portable", the context has to be narrowed to what it "does do" (and not "what it does not do").

I'd be good with removing this section altogether.

Reply:
Let me give a little bit of history of this section.  We wanted to document the minimum version of each of our dependencies.  We didn't want to lock applications into these specific versions.  If developers used these defined minimum versions, then the resulting MP applications would be portable between implementations.  But, if a newer version of one of these dependencies were used, all bets were off.  Depending on the changes introduced by this newer version, the application may or may not be portable.

For example, the defined dependency for CDI is CDI 2.0.  If CDI 2.0 is used by an application, then the app should be portable to other MP implementations.  But, if CDI 2.1 is used by an application, the app may or may not be portable.

That was the intent of this section.  We wanted to defined the minimum version for all of our dependencies, but not lock developers into these specific versions.  The caveat to this approach is that we don't know what the newer versions of the dependencies would introduce, so we can't claim portability with the unknown.

Reply:
I think we are better off not to assert either portable or non-portable. Removing the added sentence talking about `portable` sounds good to me.

Reply:
I tend to agree with @Emily-Jiang here: since MicroProfile has no notion of portability—you can't write a MicroProfile application and "run" or "deploy" it on any two MicroProfile implementations, unmodified—it probably shouldn't be mentioned here (or elsewhere).  (You can of course "port" your knowledge of and familiarity with the various APIs from one runtime to another as you re-assemble your application dedicated to that runtime, but that's different.)  To the extent that it is collectively believed that MicroProfile _should_ have some kind of notion of portability, that should be addressed as a separate (probably enormous 😄 ) effort.  OK, done hijacking this thread.

Reply:
I'm ok to remove this.